### PR TITLE
add proxy to 'middleware' in API gateway

### PR DESCRIPTION
**Pull request by IvayloPapazov**
In scope of: https://www.pivotaltracker.com/story/show/167038727

### CODE REVIEW

File: test/test.tf [hunk 1]
```diff
@@ -57,4 +58,5 @@ module "test_gateway" {
   api_aliases     = "${local.api_aliases}"
   certificate_arn = "${aws_acm_certificate_validation.cert.certificate_arn}"
   lb_fqdn         = "${local.lb_fqdn}"
+  mdw_fqdn        = "${local.mdw_fqdn}"
```
Comment (venimus) [inline test/test.tf:61]: You need to have this domain added in the certificate_arn

Comment (dincho) [reply test/test.tf:61]: no

File: main.tf [hunk 2]
```diff
@@ -14,6 +14,18 @@ resource "aws_cloudfront_distribution" "api_gate" {
...
   }
 
+  origin {
+    domain_name = var.mdw_fqdn
+    origin_id   = "api_mdw"
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
```
Comment (dincho) [inline main.tf]: This is external service so https is required, https-only ?

File: main.tf [hunk 3]
```diff
@@ -35,6 +47,28 @@ resource "aws_cloudfront_distribution" "api_gate" {
...
 
+  ordered_cache_behavior {
+    path_pattern    = "/middleware/*"
+    allowed_methods = ["HEAD", "DELETE", "POST", "GET", "OPTIONS", "PUT", "PATCH"]
+    cached_methods  = ["GET", "HEAD"]
+
+    forwarded_values {
+      query_string = true
+
+      cookies {
```
Comment (dincho) [inline main.tf:58]: We don't need cookies forward for this origin, AFAIK

File: main.tf [hunk 4]
```diff
@@ -14,6 +14,18 @@ resource "aws_cloudfront_distribution" "api_gate" {
...
 
+  origin {
+    domain_name = var.mdw_fqdn
+    origin_id   = "api_mdw"
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["TLSv1.2"]
```
Comment (dincho) [inline main.tf:25]: it also supports 1.3, which is better

Comment (IvayloPapazov) [reply main.tf:25]: 0c1083a

File: main.tf (hunk 3 above)
Comment (IvayloPapazov) [reply main.tf:58]: ad1a768

File: main.tf (hunk 2 above)
Comment (IvayloPapazov) [reply main.tf]: 0c1083a

File: test/test.tf [hunk 5]
```diff
@@ -2,7 +2,7 @@ locals {
   api_domain  = format("gw%s%s", var.envid, var.domain_sfx)
   api_aliases = [format("alias%s%s", var.envid, var.domain_sfx)]
   lb_fqdn     = format("lb%s%s", var.envid, var.domain_sfx)
-  mdw_fqdn    = format("mdw%s%s", var.envid, var.domain_sfx)
+  mdw_fqdn    = "test-middleware.test"
```
Comment (dincho) [inline test/test.tf]: if we use testnet.mdw.aepps.com we can even health check the module ?

Comment (IvayloPapazov) [reply test/test.tf]: 3cf4e62

**Review by venimus — DISMISSED** [review]

**Review by dincho — APPROVED** [review]

**Review by venimus — APPROVED** [review]

**Pull request merged**